cpp/include/cudf/hashing/detail/murmurhash3_x86_32.cuh

```diff
@@ -24,157 +24,82 @@
 #include <cudf/structs/struct_view.hpp>
 #include <cudf/types.hpp>
 
-#include <cstddef>
+#include <cuco/hash_functions.cuh>
+#include <cuda/std/cstddef>
 
 namespace cudf::hashing::detail {
 
-// MurmurHash3_x86_32 implementation from
-// https://github.com/aappleby/smhasher/blob/master/src/MurmurHash3.cpp
-//-----------------------------------------------------------------------------
-// MurmurHash3 was written by Austin Appleby, and is placed in the public
-// domain. The author hereby disclaims copyright to this source code.
-// Note - The x86 and x64 versions do _not_ produce the same results, as the
-// algorithms are optimized for their respective platforms. You can still
-// compile and run any of them on any platform, but your performance with the
-// non-native version will be less than optimal.
 template <typename Key>
-struct MurmurHash3_x86_32 {
-  using result_type = hash_value_type;
+struct MurmurHash3_x86_32 : public cuco::murmurhash3_32<Key> {
+  using result_type = typename cuco::murmurhash3_32<Key>::result_type;
 
-  constexpr MurmurHash3_x86_32() = default;
-  constexpr MurmurHash3_x86_32(uint32_t seed) : m_seed(seed) {}
-
-  [[nodiscard]] __device__ inline uint32_t fmix32(uint32_t h) const
-  {
-    h ^= h >> 16;
-    h *= 0x85ebca6b;
-    h ^= h >> 13;
-    h *= 0xc2b2ae35;
-    h ^= h >> 16;
-    return h;
-  }
-
-  [[nodiscard]] __device__ inline uint32_t getblock32(std::byte const* data,
-                                                      cudf::size_type offset) const
+  __host__ __device__ constexpr MurmurHash3_x86_32(uint32_t seed = cudf::DEFAULT_HASH_SEED)
+    : cuco::murmurhash3_32<Key>{seed}
   {
-    // Read a 4-byte value from the data pointer as individual bytes for safe
-    // unaligned access (very likely for string types).
-    auto const block = reinterpret_cast<uint8_t const*>(data + offset);
-    return block[0] | (block[1] << 8) | (block[2] << 16) | (block[3] << 24);
   }
 
-  [[nodiscard]] result_type __device__ inline operator()(Key const& key) const
+  __device__ result_type operator()(Key const& key) const
   {
-    return compute(normalize_nans_and_zeros(key));
+    return cuco::murmurhash3_32<Key>::operator()(key);
   }
 
-  template <typename T>
-  result_type __device__ inline compute(T const& key) const
+  template <typename Extent>
```

Comment:
We should use `std::uint64_t` here and remove the template instantiation. It's unnecessary as all byte ranges will be within that range.